app/models/king.rb

```diff
@@ -1,8 +1,12 @@
 class King < Piece
-  # ensures king piece can only move 1 square in any direction
+
   def valid_move?(x,y)
-    # checks critera for valid_move? that apply to all pieces and returns false if fails
-    return false unless super(x,y)
+    # checks critera for valid_move? that apply to all pieces and returns false if fails unless can_castle
+    (normal_valid_move?(x,y) && super(x,y)) || can_castle?(x,y)
```

Comment:
The order should switch so that `super(x, y)` is called before `normal_valid_move?` as that preserves what happened before:

```ruby
(super(x, y) && normal_valid_move?(x,y)) || can_castle?(x,y)
```

You could, BTW, keep the old `valid_move?` and just add the castle check as another condition:

```ruby
  def valid_move?(x, y)
    # returns true if moving only 1 square any direction
    return true if (x_diff(x) <= 1) && (y_diff(y) <= 1)

    # returns true if the king can castle
    return true if can_castle?(x, y)

    return false
  end
```